pkgs/by-name/ze/zensical/package.nix

```
@@ -0,0 +1,69 @@
+{
+  lib,
+  fetchFromGitHub,
+  runCommand,
+  rustPlatform,
+  python3Packages,
+  versionCheckHook,
+}:
+
+python3Packages.buildPythonApplication rec {
+  pname = "zensical";
+  version = "0.0.5";
+  pyproject = true;
+
+  src =
+    let
+      zensical = fetchFromGitHub {
+        owner = "zensical";
+        repo = "zensical";
+        tag = version;
+        hash = "sha256-tEHY4O+odAhFvLentogJPUqT+xYVi8z0HXCml0A3Adw=";
+      };
+      ui = fetchFromGitHub {
+        owner = "zensical";
+        repo = "ui";
+        rev = "3fe06b9ce12f4e7142d34d83cf3085d345318a43";
+        hash = "sha256-CXj+C8GvBVrydUp8GjkX1857msclknHSXqyJ+BoBQvU=";
+      };
+    in
+    runCommand "combined" { } ''
+      mkdir -p $out
+      cp -r ${zensical}/. $out/
+
+      chmod u+w $out/python/zensical
+      mkdir $out/python/zensical/templates
+      cp -r ${ui}/dist/. $out/python/zensical/templates
+    '';
+
+  cargoDeps = rustPlatform.fetchCargoVendor {
+    inherit src;
+    hash = "sha256-ljNoAaqXAVbnF/9RbXdWvJAac8sh0W3v53YtgiNQEng=";
+  };
+
+  nativeBuildInputs = [
+    rustPlatform.maturinBuildHook
+    rustPlatform.cargoSetupHook
+  ];
+
+  propagatedBuildInputs = with python3Packages; [
+    click
+    deepmerge
+    markdown
+    pygments
+    pymdown-extensions
+    pyyaml
+  ];
+
+  nativeCheckInputs = [ versionCheckHook ];
+  versionCheckProgramArg = "--version";
+
```

a useful addon (especially since this app is gonna get frequent updates)
```suggestion
  passthru.updateScript = nix-update-script { };
```